profiling/plugins/log_plugin/log_plugin.c

```diff
@@ -0,0 +1,142 @@
+#include "log_plugin.h"
+
+#include <SAPI.h>
+#include <Zend/zend.h>
+#include <pthread.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+typedef datadog_php_logger logger_t;
+typedef datadog_php_log_level log_level_t;
+typedef datadog_php_string_view string_view_t;
+
+static logger_t profiler_logger = DATADOG_PHP_LOGGER_INIT;
+static pthread_mutex_t logger_mutex;
+
+// Use noop log pattern to avoid initialization issues {{{
+static void noop_log(datadog_php_log_level ll, datadog_php_string_view sv) {
+  (void)ll, (void)sv;
+}
+
+static int64_t noop_logv(datadog_php_log_level level, size_t n_messages,
+                         datadog_php_string_view messages[static n_messages]) {
+  (void)level, (void)n_messages, (void)messages;
+  return -1;
+}
+
+void noop_log_cstr(datadog_php_log_level log_level, const char *cstr) {
+  (void)log_level, (void)cstr;
+}
+
+datadog_php_static_logger prof_logger = {
+    .log = noop_log,
+    .logv = noop_logv,
+    .log_cstr = noop_log_cstr,
+}; // }}}
+
+static void datadog_php_log_plugin_log(log_level_t log_level,
+                                       string_view_t message) {
+  datadog_php_log(&profiler_logger, log_level, message);
+}
+
+/**
+ * @param log_level
+ * @param cstr Must not be null.
+ */
+static void datadog_php_log_plugin_log_cstr(datadog_php_log_level log_level,
+                                            const char *cstr) {
+  string_view_t message = {strlen(cstr), cstr};
+  datadog_php_log(&profiler_logger, log_level, message);
+}
+
+static int64_t datadog_php_log_plugin_logv(
+    datadog_php_log_level level, size_t n_messages,
+    datadog_php_string_view messages[static n_messages]) {
+  return datadog_php_logv(&profiler_logger, level, n_messages, messages);
+}
+
+// todo: extract this to a plugin?
+#if PHP_VERSION_ID >= 80000
+#define sapi_getenv_compat(name, name_len) sapi_getenv((name), name_len)
+#elif PHP_VERSION_ID >= 70000
+#define sapi_getenv_compat(name, name_len) sapi_getenv((char *)(name), name_len)
+#else
+#define sapi_getenv_compat(name, name_len)                                     \
+  sapi_getenv((char *)(name), name_len TSRMLS_CC)
+#endif
+
+static ZEND_COLD void datadog_php_log_plugin_init(void) {
+  int descriptor = dup(STDERR_FILENO);
+  if (descriptor < 1) {
+    return;
+  }
+
+  pthread_mutexattr_t mutex_attr;
+  if (pthread_mutexattr_init(&mutex_attr) != 0) {
+    goto startup_descriptor_cleanup;
+  }
+  if (pthread_mutexattr_settype(&mutex_attr, PTHREAD_MUTEX_ERRORCHECK) != 0) {
+    goto startup_descriptor_cleanup;
+  }
+  if (pthread_mutex_init(&logger_mutex, &mutex_attr) != 0) {
+    goto startup_descriptor_cleanup;
+  }
+
+  string_view_t env_var =
+      datadog_php_string_view_from_cstr("DD_PROFILING_LOG_LEVEL");
```

Comment:
Did we ever end up figuring out if we're OK with the name being `profiling` (instead of `profiler` to be more similar to `tracer`)? Are other languages also going with `profiling` vs `profiler`? (BTW This is not something we need to figure out before merge.)

Reply:
`DD_PROFILING_` is the standardized env var prefix for profiling. I have asked the team for more context and will relay what I hear.